# CSE110 Assignment 2 

This assignment is to guide in solving basic programming problems using loop in Python. 




##Write your name, student ID and theory section number below:


In [ ]:
#STUDENT NAME: Tahia Hossain
#STUDENT ID: 21301164
#CSE110 SECTION: 02

## Write the code in Python for following problems:

**<font color='red'>[MUST MAINTAIN VARIABLE NAMING CONVENTIONS FOR ALL THE TASKS]</font>**

### Task 1

Write python program, which prints the following sequences of values in loops:

a)	24, 18, 12, 6, 0, -6\
b)	-10, -5, 0, 5, 10, 15, 20\
c)	18, 27, 36, 45, 54, 63\
d)	18,-27, 36,-45,54,-63

=====================================================================

**Hints(1):** Use a while loop for solving these problems.

**Hints(2):** We are already familiar with the print() function. But when we use it to print any value it automatically adds an additional newline after each print statement.

For example:\
print(1)\
print(2)

Output:\
1\
2

=====================================================================

To solve this problem, in Python3, we need to add an extra argument (end = " ") in the print function which tells the program to skip printing the additional newline.

For example:\
print(1, end =" ")\
print(2)

Output:(prints the next output right to the previous one)\
12

=====================================================================

In Task-1(a), the loop counter should be initialized at 24 and the loop should terminate when the loop counter reaches -6. The difference between the first two values is 24-18=6. So the loop counter value is getting decremented by 6 in every iteration.

For your understanding task 1(a) code is done for you.


In [ ]:
# a) 24, 18, 12, 6, 0, -6

# initialize loop counter
counter = 24

# loop structure
while counter >= -6:
    
    #inside loop body
    if counter == -6:
        print(counter, end = "") 
    else:
        print(counter, end = ", ") 
     
    counter = counter - 6 #updating loop counter
    #inside loop body
    
#outside loop body

24, 18, 12, 6, 0, -6

In [ ]:
# b)-10, -5, 0, 5, 10, 15, 20

counter=-10
while (counter<=20) :
    if counter==20 :
        print(counter)
        break
    print(counter,end = ', ')
    counter = counter + 5

-10, -5, 0, 5, 10, 15, 20


In [ ]:
# c)18, 27, 36, 45, 54, 63

counter=18
while (counter<=63) :
    if counter==63 :
        print(counter)
        break
    print(counter,end = ', ')
    counter = counter + 9

18, 27, 36, 45, 54, 63


**Hints for 1(d):** 

print(5 * (-1)) gives output -5\
print("-" + str(5)) gives output -5

In [ ]:
# d)18,-27, 36,-45,54,-63

start = 18
while start<=63 :
    if start == 63 :
        print(start*-1)
        break
    elif start%2 == 0 :
        print(start, end = ',')
    else :
        print(start*-1, end = ',')
    start = start + 9

18,-27,36,-45,54,-63


====================================================================\
<font color='blue'>
All the tasks can be solved using both the "while loop" and the "for loop". But you need to choose the best looping construct which is the most suitable for solving a particular task.
</font>

====================================================================

### Task 2
Write a Python code for the following:

1)	Ask the user to enter the name of his favorite car.\
2)	Ask the user to enter a Number

Display the name of the user’s favorite car, the number of times specified in the second step.

==========================================================

**Example 01:** If the user enters “Toyota” and 20, your program should print the name “Toyota” twenty times.

**Input:**\
Toyota\
2

**Output:**\
Toyota\
Toyota

==========================================================

**Example02:** If the user enters “Veyron” and 5, your program should print the name “Veyron” five times.

**Input:**\
Veyron\
5

**Output:**\
Veyron\
Veyron\
Veyron\
Veyron\
Veyron


In [ ]:
#Todo

name=input("Please enter the name of your favorite car: ")
number=int(input("Please enter a Number: "))

count = 0
while (count<number):
    print(name)
    count+=1

Please enter the name of your favorite car: Limousine
Please enter a Number: 3
Limousine
Limousine
Limousine


### Task 3
Write a Python code of a program that adds all numbers that are multiples of **both 7 and 9** up to 600 (including 600).

Output: 2835


In [ ]:
#Todo

number = 0
summation=0
while number<=600 :
    if number%7 == 0 and number%9 == 0 :
        summation = summation + number
    number = number+1
print(summation)

2835


### Task 4
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9** up to 600(including 600). \
Ensure that numbers like 63 are added only once in the sum.

Output: 42649


In [ ]:
#Todo

number = 0
sum=0
while number<=600 :
    if number%7 == 0 or number%9 == 0 :
        sum = sum + number
    number = number+1
print(sum)

42649


### Task 5
Write a Python code of a program that adds all numbers that are multiples of **either 7 or 9 but not both**, up to 600(including 600).

Output: 39814


In [ ]:
#Todo

sum=0
for number in range (1,601) :
  if (number%7 == 0 or number%9 == 0) and not (number%7 == 0 and number%9 == 0) :
      sum = sum + number
print(sum)

39814


### Task 6
Write a Python code to displays all the **odd numbers** between 10 and 50.

Output: 11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49


In [ ]:
#Todo

for number in range (10,51) :
    if number%2 != 0 :
        print(number, end = ' ')

11 13 15 17 19 21 23 25 27 29 31 33 35 37 39 41 43 45 47 49 

### Task 7
Write a Python code that will calculate the **value of y if the expression** of y is as follows (n is the input):  

$y = 1^2 - 2^2 +3^2 -4^2 +5^2 .........+ n^2 $


==========================================================

**Example01:** 

**Input:**\
10

**Output:**\
-55

==========================================================

**Example02:** 

**Input:**\
20

**Output:**\
-210



In [ ]:
#Todo

odd = 0
even = 0
n = int(input("Please enter a number: "))

for index in range (1,n+1) :
    if index%2 != 0 :
        odd = odd + index**2
    else :
        even = even + index**2
y = odd - even
print("The value of y is:", y)

Please enter a number: 10
The value of y is: -55


### Task 8

Write a Python code of a program that asks the user to enter ten numbers then display the total and the average of **ONLY** the **odd numbers** among those ten numbers. <font color='red'>[Use of list is completely unnecesary for this task]</font>

==========================================================

**Example01:** 

**Input:**\
1

2

3

4

5

6

7

8

9

10

**Output:**
Total is 25 and Average is 5.0


**Explanation:**

Total is 25 = (1+3+5+7+9) and Average is 25/5 = 5.0

==========================================================


**Example02:** 

**Input:**\
-20

3

-5

40

-17

9

20

-8

99
  
-200

**Output:**

Total is 89 and Average is 17.8


**Explanation:**

Total is 89 =(3+(-5)+(-17)+9+99) and Average is 89/5 = 17.8



In [ ]:
#Todo

sum = 0
odd_count = 0

for iteration in range (0,10) :
  number = int(input("Please enter a number: "))
  if number%2 == 1 :
    sum += number
    odd_count += 1
average = sum/odd_count
print("Total is", sum, "and Average is", average)

Please enter a number: 1
Please enter a number: 2
Please enter a number: 3
Please enter a number: 4
Please enter a number: 5
Please enter a number: 6
Please enter a number: 7
Please enter a number: 8
Please enter a number: 9
Please enter a number: 10
Total is 25 and Average is 5.0


### Task 9
Write a Python code for the following:
* Ask the user to enter a Number, N
* Display the summation of multiples of 7 up to that number (**from 1 to N inclusive**)

==========================================================

**Example01:** 

**Input:**\
50

**Output:**\
196

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 = 196

==========================================================

**Example02:** 

**Input:**\
75

**Output:**\
385

**Explanation:**\
7 + 14 + 21 + 28 + 35 + 42 + 49 + 56 + 63 + 70 = 385





In [ ]:
#Todo

number=int(input("Please enter a number, N: "))
count=0
sum=0

#the problem can also be done by using for loop
#doing the other way for later benefit

#for i in range (1,number+1) :
#if i%7 == 0 :
#sum += i
#print(sum)

while count<number:
  if count%7 == 0 :
    sum = sum + count
    count = count + 1
  else :
    count = count + 1
print(sum)

Please enter a number, N: 50
196


### Task 10
Write a Python code that will read 10 numbers from the user, and then print the first number, the sum of the first 2 numbers, the sum of the first 3 numbers, and so on up to the sum of 10 numbers.

==========================================================


For example,

The user enters 10, output 10, then \
The user enters 2, (10+2) = 12, output 12, then \
The user enters 4, (10+2+4) = 16, output 16, then \
The user enters 20, (10+2+4+20) = 36, output 36, and continues till the 10$^{th}$ input


In [ ]:
#Todo

number_1=int(input("Please enter the first number: "))
print(number_1)
sum = number_1

count = 1
while count<10 :
  user_input=int(input("Please enter another number: "))
  sum = sum + user_input
  print(sum)
  count = count + 1

Please enter the first number: 10
10
Please enter another number: 2
12
Please enter another number: 4
16
Please enter another number: 20
36
Please enter another number: 1
37
Please enter another number: 2
39
Please enter another number: 3
42
Please enter another number: 4
46
Please enter another number: 5
51
Please enter another number: 6
57





### Task 11
Write a Python program which takes a number and prints the digits from the unit place, then the tenth, then hundredth, etc. <font color='red'>(Right to Left)</font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

**Example:** If the user gives 32768, then print 8, 6, 7, 2, 3 

=========================================================================

**Hint (1):**  The input() function, converts the input data to String data type by default. Use this knowledge to solve this problem.


**Hint  (2):** First to get the digit from the right side, take the remainder of the number using modulus (%) operator and print the digit. For dropping the last digit, perform floor division by 10 on the number. Then start over.

32,768 % 10 = 8 

32,768 // 10 = 3, 276 

3, 276 % 10 = 6

3, 276 // 10 = 327 

327 % 10 = 7 

327 // 10 = 32 

32 % 10 = 2 

32 //10 = 3 

3 % 10 = 3 

3 // 10 = 0 

Done! When the number becomes 0 your loop should end.



In [ ]:
#Todo

number=int(input("Please enter a number: "))

while number > 0 :
  remainder = number%10
  number = number // 10
  if number == 0 :
    print(remainder, end = "")
  else :
    print(remainder, end = ', ')

Please enter a number: 231123
3, 2, 1, 1, 3, 2

### Task 12

Write a Python program that takes a number and tells how many digits are in that number. \
<font color='red'>[Consider the input number to be an INTEGER.]</font>

<font color='red'>You are not allowed to use len() function</font>

**Example:** If the user gives 9876, you should print 4. 



**Hint:** Keep floor dividing by ten and count how many times this could be divided. 

9876 floor divide by 10, is 987, count that got 1 digit (total 1) \
987 floor divide by 10, is 98, count that got 1 digit (total 2) \
98 floor divide by 10, is 9, count that got 1 digit (total 3) \
9 floor divide by 10, is 0, count that got 1 digit (total 4) 

Done! (When the number becomes 0 your loop should end.)



In [ ]:
#Todo

digit = 0
number=int(input("Please enter a number: "))

while number != 0 :
  number = number // 10
  digit = digit + 1
print(digit)

Please enter a number: 12345
5


### Task 13
Write a Python program that takes a number from the user and prints its digits from left to right. <font color='red'> (left to right) </font>

<font color='red'>[Consider the input number to be an INTEGER. You are not allowed to use String indexing for solving this task]</font>

=========================================================================

**Example:** if the user gives 32768, then print 3, 2, 7, 6, 8 

=========================================================================

**Hint(1):**  The input() function, converts the input data to String data type by default. Use this knowledge to solve this problem.

**Hint(2):**  

Step1: count how many digits\
Step2: calculate 10 to the power that (number of digits) minus 1.

Step3: Say, 32768 has 5 digits, so you calculate 10 to the power 4 which is 10,000. Then divide 32,768 by 10,000 and thus you get 3.\
Take remainder of 32,768 by 10,000 and thus you get 2,768 Then divide 10,000 by 10 to get 1,000\
Then divide 2,768 by 1,000 and thus you get 2.\
take remainder of 2,768 by 1,000 and thus you get 768 keep going on until there is no more digits left (zero!).


**In short:**\
Loop 1: First count digits, say 5 in this case for 32,768\
Loop 2: Then calculate 10 to the power 4 (5-1), that is 10,000.\
Loop 3: Then repeat following three steps\
32,768 // 10,000 = 3\
32,768 % 10,000 = 2,768 \
10,000//10 = 1,000

2,768 // 1,000 = 2\
2,768 % 1,000 = 768\
1,000//10 = 100

768 // 100 = 7\
768 % 100 = 68 \
100//10= 10

68 //10 = 6\
68 % 10 = 8 \
10//10= 1

8 //1 = 8\
8 % 1 = 0 \
1//10=0


In [ ]:
#Todo

count=0
number=int(input("Please enter a number: "))
cache = number

while number>0 :
  number = number // 10
  count += 1

while cache>0 :
  if cache%10**(count-1) == 0 :
    figure = cache // 10**(count-1)
    print(figure)
    break
  else :
    figure = cache // 10**(count-1)
    cache = cache%10**(count-1)
    count -=1
    print(figure, end = ", ")

Please enter a number: 23456
2, 3, 4, 5, 6


### Task 14

Write a Python program that takes a number from the user and prints the divisors of that number and then print how many divisors were there.
<font color='red'>[The input number has to be an INTEGER]</font>

==========================================================

**Example01:** 

**Input:**\
6

**Output:**\
1, 2, 3, 6 \
Total 4 divisors.


==========================================================

**Example02:** 

**Input:**\
121

**Output:**\
1, 11, 121 \
Total 3 divisors.




In [ ]:
#Todo

number=int(input("Please enter a number: "))
count = 0

for divisor in range (1,number+1) :
  if divisor == number :
    print(divisor)
    break
  elif number%divisor == 0 :
    print(divisor, end = ", ")
    count += 1

print('Total', count+1, 'divisors.')

Please enter a number: 6
1, 2, 3, 6
Total 4 divisors.


### Task 15
Write a Python program that takes a number and tells if it is a perfect number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Perfect Number:** An integer number is said to be a perfect number if its factors, including 1 but not the number itself, sum to the number.

==========================================================

**Example01:** 

**Input:**\
6

**Output:**\
6 is a perfect number

**Explanation:**\
6 have 4 divisors: 1, 2, 3, and 6. \
If we add all factors except itself, 6 = 1 + 2 + 3.

==========================================================


**Example02:** 

**Input:**\
28

**Output:**\
28 is a perfect number

**Explanation:**\
28 have 6 divisors: 1, 2, 4, 7, 14, and 28. \
If we add all factors except itself, 28 = 1 + 2 + 4 + 7 + 14.

==========================================================

**Example03:** 

**Input:**\
33

**Output:**\
33 is not a perfect number

**Explanation:**\
33 have 3 divisors: 1, 3, 11, and 33. \
If we add all factors except itself, 15 = 1 + 3 + 11.



In [ ]:
#Todo

number=int(input("Please enter a number: "))
sum=0

for iteration in range (1,number) :
  if number%iteration == 0 :
    sum += iteration
if sum == number :
  print(number, "is a perfect number")
else :
  print(number, "is not a perfect number")

Please enter a number: 33
33 is not a perfect number


### Task 16

Write a Python program that asks the user for one number and tells if it is a prime number or not. <font color='red'>[The input number has to be an INTEGER]</font>

**Prime Number:** If a number has only two divisors, (1 and itself), then it is prime. If it is divisible by more numbers, then it is not a prime.

==========================================================

**Hint:** use the divisor count from task 16. 

==========================================================

**Example01:** 

**Input:**\
11

**Output:**\
11 is a prime number

**Explanation:**\
11 has only 2 divisors: 1, and 11.

==========================================================

**Example02:** 

**Input:**\
6

**Output:**\
6 is not a prime number

**Explanation:**\
6 have 4 divisors: 1, 2, 3 and 6.






In [ ]:
#Todo

user_input=int(input("Please enter a number: "))
count=0

for number in range (1,user_input+1):
  if user_input%number == 0 :
    count += 1
if count == 2 :
  print(user_input, "is a prime number")
else :
  print(user_input, "is not a prime number")

Please enter a number: 23
23 is a prime number


### Task 17

Write a Python program that asks the user for a quantity then takes that many numbers and prints the maximum, minimum and average of those numbers. 

**<font color='red'>[CANNOT USE MAX,MIN BUILT-IN FUNCTIONS][FOR THIS TASK YOU DO NOT NEED TO USE LISTS]</font>** 

==========================================================

**Example01:** If the user enters 5 as an input for quantity. Then enters 10, 4, -1, -100, and 1. \
Your program output should be: “Maximum 10”, “Minimum -100”, “Average is -17.2”

**Input:**\
5
10
4
-1
-100
1.

**Output:**\
Maximum 10\
Minimum -100\
Average is -17.2

**Explanation:**\
Average calculation: (10+4 + (-1) + (-100) + 1)/5 = -86/5=-17.2



In [ ]:
#Todo

user_input=int(input("Please enter a number: "))
total=0
min=0
max=0

for iteration in range (0,user_input) :
  number=int(input())

  if iteration == 0 :
    min=number
    max=number
    total=number
  else :
    if number<min :
      min=number
    elif number>max:
      max=number
    total = total + number
average = total / user_input
print("Maximum", max)
print("Minimum", min)
print("Average is", average)

Please enter a number: 5
10
4
1
-1
-100
Maximum 10
Minimum -100
Average is -17.2


## Task 18

Write a python program that prints a square of size N using + where N will be given as input.

=======================================

**Sample Input**
5

**Sample Output**<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>
+++++<br>


=======================================

**Sample Input**
3

**Sample Output**<br>
+++<br>
+++<br>
+++<br>

In [ ]:
#Todo

number=int(input("Please enter a number: "))

for length in range (number) :
  for breadth in range (number) :
    print("+", end = "")
  print()

Please enter a number: 5
+++++
+++++
+++++
+++++
+++++


### Task 19

Write a python program that prints a rectangle of size M (height/line numbers) and N(length/column numbers) using incrementing numbers where M,N will be given as input.

===================================

**Sample Input**
4,6

**Sample Output**<br>
123456<br>
123456<br>
123456<br>
123456<br>


===================================


**Sample Input**
3,2

**Sample Output**<br>
12<br>
12<br>
12<br>


In [ ]:
#Todo

given_number=input("Please enter height and length separated by ',' (like 3,2): ")

M,N=given_number.split(',')
M=int(M)
N=int(N)

for height in range (0,M) :
  for length in range (0,N) :
    print(length+1, end = "")
  print()

Please enter height and length separated by ',' (like 3,2): 4,6
123456
123456
123456
123456


### Task 20

Write a python program that prints a right angled triangle of height N using incrementing numbers where N will be given as input.

====================================

**Sample Input**
4

**Sample Output**<br>
1<br>
12<br>
123<br>
1234<br>

====================================

**Sample Input**
5

**Sample Output**<br>
1<br>
12<br>
123<br>
1234<br>
12345<br>

In [ ]:
#Todo

user_input=int(input("Please enter a number, N: "))

for height in range (user_input) :
  for incrementing_number in range (height+1) :
    print(incrementing_number+1, end = "")
  print()

Please enter a number, N: 5
1
12
123
1234
12345


## Optional Tasks (21 -26) [Ungraded]

### Task 21

Write a python program that prints all the fibonacci number from 0 to N where N will be given. 

**A Fibonacci number is a number which is the summation of its previous two fibonacci number.**

First two fibonacci number are 0 and 1. So the 3rd Fib will be 0+1=1, 4th Fib will be 1+1=2, 5th Fib will be 1+2=3 and so on.

==============================

**Sample Input**<br>
10

**Sample Output**<br>
0 1 1 2 3 5 8

==============================


**Sample Input**<br>
15

**Sample Output**<br>
0 1 1 2 3 5 8 13

In [ ]:
#Todo

user_input=int(input("Please enter a number: "))
number_1=0
number_2=1

print("0 1", end = "")

Please enter a number: 1
0 1

### Task 22

Write a python program that converts a Decimal Integer number to a Binary Number. 

**A decimal can be converted to a binary number by keeping track of the remainders after each division of that number by 2. **

================================

For example, to convert 10 to a binary number, we can follow the following approach

10/2 = 5 (Remainder 0) 
<br>5/2 = 2 (Remainder 1) 
<br>2/2 = 1 (Remainder 0) 
<br>1/2 = 0 (Remainder 1) 

Take the remainders from bottom to top, which is, 1010. Binary of 10 is 1010.

**Sample Input** <br>
10

**Sample Output**<br>
1010

================================

For example, to convert 13 to a binary number, we can follow the following approach

13/2 = 6 (Remainder 1) 
<br>6/2 = 3 (Remainder 0) 
<br>3/2 = 1 (Remainder 1) 
<br>1/2 = 0 (Remainder 1) 

Take the remainders from bottom to top, which is, 1101. Binary of 13 is 1101.

**Sample Input** <br>
13

**Sample Output**<br>
1101

In [ ]:
#Todo

### Task 23

Write a python program that converts a Binary number to a Decimal Integer Number. 

**A binary number can be converted to its corresponding decimal number by multiplying each binary digit with a power of 2 where the power denotes the position of the binary digit**

========================================

For example, to convert 1010, we can do this -


$2^{3}$ x 1 + $2^{2}$ x 0 + $2^{1}$ x 1 + $2^{0}$ x 0 = 10


**Sample Input** <br>
1010

**Sample Output** <br>
10

========================================

For example, to convert 1101, we can do this -


$2^{3}$ x 1 + $2^{2}$ x 1 + $2^{1}$ x 0 + $2^{0}$ x 1 = 13


**Sample Input** <br>
1101

**Sample Output** <br>
13

In [ ]:
#Todo

user_input=input("Please enter a number: ")


### Task 24

Take five numbers from the user and find the minimum and the average of **only the even** numbers entered by the user. [If the user enters odd numbers ignore them] 

**<font color='red'>[CANNOT USE MAX,MIN BUILT-IN FUNCTIONS] [FOR THIS TASK YOU DO NOT NEED TO USE LISTS]</font>** 

Assume, the first input is always an even number.

============================================================

**Example01:** If the user enters 10, 4, -1, -100, and 1. \
Output: “Minimum -100”, “Average is 28.66667”

**Input:**\
10

4

-1

-100

1

**Output:**\
Minimum -100\
Average is 28.66667

**Explanation:**\
Average calculation: (10+4 + (-100))/3 = -86/3=-28.66667

============================================================

**Example02:** If the user enters 2, 10, 1, 21, and 3. \
Output: Minimum 2”, “Average is 6.0”

**Input:**\
2

10

1

21

3

**Output:**\
Minimum 2\
Average is 6.0

**Explanation:**\
Average calculation: (2+10)/2 = 12/2= 6





In [ ]:
#Todo

counter=0
add=0
count_even=0
min=0

while counter < 5 :
  given_input=int(input("Please enter a number: "))
  if given_input%2 == 0 :
    count_even = count_even + 1
    add += given_input
  if counter == 0 :
    min = given_input
  elif given_input < min :
    max = given_input
  counter = counter + 1
print("Minimum", min)
print("Average is", add/count_even)

Please enter a number: 2
Please enter a number: 10
Please enter a number: 1
Please enter a number: 21
Please enter a number: 3
Minimum 2
Average is 6.0


### Task 25

Write a Python code for the following:
* Ask the user to enter a number, N (Total number of inputs to be taken). Assume that the user will never enter the first number as zero.
* Take N number of inputs
* Prints out the product of all the numbers read

For example, if the first input is 4, then the program has to read in four numbers from the user and print the product of these four numbers. 

=========================================================================

**Example01:**

**Input:**\
5 

10 

6 

4 

2 

1 

**Output:**\
480

**Explanation:**\
5 is the total number of inputs taken. Then, the calculation should be 10 X 6 X 4 X 2 X 1 = 480 







In [ ]:
#Todo

user_input_n=int(input("Please enter a number: "))
counter = 0
multiplication = 1

while counter<user_input_n :
  number=int(input("Please enter a number: "))
  multiplication *= number
  counter = counter + 1
print(multiplication)

Please enter a number: 5
Please enter a number: 10
Please enter a number: 6
Please enter a number: 4
Please enter a number: 2
Please enter a number: 1
480


### Task 26
Ask the user for a range (a starting number and an ending number). 
**Count** how many numbers are prime numbers and how many numbers are perfect numbers between that range and **print** those numbers. 


=========================================================================

**Hint (1):** Declare two strings/lists to store the prime and perfect numbers. Inside the iteration store the values of the prime and perfect numbers in the pre- declared variables.

**Hint (2):**


ans = "Prime numbers: "\
prime_value = 11
ans = ans + str(prime_value) + " "\
print(ans)

ans = ans + str(13) + " "\
print(ans)

*Output:*

Prime numbers: 11 \
Prime numbers: 11 13 

=========================================================================

**Example:**between 2 and 6, there are 3 prime numbers (2, 3, 5) and 1 perfect number (6). 

**Input:**\
2

6

**Output:**\
Between 2 and 6,\
Found 3 prime numbers \
Found 1 perfect number\
Prime numbers: 2  3  5 \
Perfect numbers: 6 



In [ ]:
#Todo
